index.bs

```diff
@@ -153,6 +156,24 @@ Note: These steps have to be run on the [=file system queue=].
 
 </div>
 
+<div algorithm>
+To <dfn for="file system entry/lock">check for a descendant lock</dfn> on a given
+[=/file system entry=] |entry|:
+
+1. Let |lock| be the |entry|'s [=file system entry/lock=].
+1. If |lock| is not "`open`":
+   1. Return true.
+1. If |entry| is a [=directory entry=]:
+   1. [=set/For each=] |child| of |entry|'s [=directory entry/children=]:
+      1. Let |hasDescendantLock| be the result of [=file system entry/lock/checking for a descendant lock=] on |child|.
+      1. If |hasDescendantLock| is true:
+         1. Return true.
+1. Return false.
```

Comment:
The name of this algorithm doesn't give any indication that this returns something, or what the return value means. It could be nice to return something that's more self-explanatory

 For example, the "take" algorithm above returns "`failure`" or "`success`". Maybe this should return "`taken`" or "`open`"?